models/icepack/model_mod.f90

```diff
@@ -0,0 +1,885 @@
+      print*,'obstype TSFC expected obs:',expected_obs
+      print*,'TSFC tsfc lat lon:' ,llat,llon
+      print*,'temp:',temp
+      print*,'temp1:',temp1
+   endif
+else
+    call lon_lat_interpolate(state_handle, ens_size, base_offset, llon, llat, set_obstype, cat_signal, expected_obs, istatus)
+    
+      if (any(expected_obs<0.0) .and. (debug > 1)) then
+      print*,'obstype SIC expected concs:',expected_obs
+      print*,'SIC sic negative lat lon:',llat,llon
+      endif
+      if (any(expected_obs>1.0) .and. (debug > 1)) then
+      print*,'obstype SIC expected concs:',expected_obs
+      print*,'SIC sic positive lat lon:',llat,llon
+      endif
+endif
+
+if (cat_signal == -1) then
+      ! we need to know the aggregate sea ice concentration for these special cases
+      base_offset = get_index_start(domain_id, get_varid_from_kind(QTY_SEAICE_CONCENTR))
+      base_offset = base_offset + (cat_index-1)
+      call lon_lat_interpolate(state_handle, ens_size, base_offset, llon, llat, set_obstype, cat_signal, expected_aggr_conc, istatus)
+      expected_obs = expected_obs/max(expected_aggr_conc,1.0e-8)  ! hope this is allowed so we never divide by zero
+
+      if ((any(expected_aggr_conc<0.0) .or. any(expected_aggr_conc>1.0)) .and. (debug > 1)) then
+      print*,'obstype SIT expected conc:',expected_aggr_conc
+      print*,'SIT sic lat lon:',llat,llon
+      endif
+
+endif
+
+if (debug > 1) print *, 'interp val, istatus = ', expected_obs, istatus, size(expected_obs)
+
+! This should be the result of the interpolation of a
+! given obs_type of variable at the given location.
+
+! The return code for successful return should be 0. 
+! Any positive number is an error.
+! Negative values are reserved for use by the DART framework.
+
+end subroutine model_interpolate
+
+!------------------------------------------------------------------
+
+subroutine lon_lat_interpolate(state_handle, ens_size, offset, lon, lat, var_type, cat_signal, expected_obs, istatus)
+type(ensemble_type), intent(in)  :: state_handle
+integer,             intent(in)  :: ens_size
+integer(i8),         intent(in)  :: offset
+real(r8),            intent(in)  :: lon, lat
+integer,             intent(in)  :: var_type
+integer,             intent(in)  :: cat_signal
+real(r8),            intent(out) :: expected_obs(ens_size)
+integer,             intent(out) :: istatus(ens_size)
+
+real(r8) :: work_expected_obs(ens_size)
+integer  :: e, iterations, Niterations
+integer(i8) :: state_index
+
+if ( .not. module_initialized ) call static_init_model
+
+istatus = 0
+if (var_type == 14) then
+  e = 1
+else if (var_type == 15) then
+  e = 2
+else if (var_type == 16) then
+  e = 3
+endif
+if ( cat_signal < 1 )  then
+   Niterations = Ncat ! only iterate if aggregating over all types
+else
+   Niterations = 1 ! no need to iterate
+endif
+work_expected_obs = 0.0_r8
+expected_obs = 0.0_r8
+do iterations = 1, Niterations
```

Comment:
the loop variable iterations is not used in this loop.

Reply:
Will be addressed in the refactor/design of the interpolation code along with other related suggestions